packages/core/src/interactions/packet/acknowledgement.ts

```diff
@@ -114,12 +114,9 @@ class PacketAcknowledgementInteraction<Chain extends HoprCoreConnector>
 
         let ticketCounter: Uint8Array
         try {
-          ticketCounter = toU8a(
-            (u8aToNumber(
-              (await this.node.db.get(Buffer.from(this.node._dbKeys.AcknowledgedTicketCounter()))) as Uint8Array
-            ) as number) + 1,
-            ACKNOWLEDGED_TICKET_INDEX_LENGTH
-          )
+          let tmpTicketCounter = await this.node.db.get(Buffer.from(this.node._dbKeys.AcknowledgedTicketCounter()))
+
+          ticketCounter = u8aAdd(true, tmpTicketCounter, ONE)
```

Comment:
so the error was we were adding 1 to a u8a? Why don't the types prevent this?

Reply:
good question ...